apps/ewallet_config/priv/repo/migrations/20181008093853_add_settings.exs

```diff
@@ -0,0 +1,27 @@
+defmodule EWalletConfig.Repo.Migrations.AddSettings do
+  use Ecto.Migration
+
+  def change do
+    create table(:setting, primary_key: false) do
+      add :uuid, :uuid, primary_key: true
+      add :id, :string, null: false
+
+      add :key, :string, null: false
+      add :data, :map
+      add :encrypted_data, :binary
+      add :type, :string, null: false
+      add :description, :string
+      add :options, :map
+      add :parent, :string
+      add :parent_value, :string
+      add :secret, :boolean, null: false, default: false
+      add :position, :integer, null: false, unique: true
+
+      timestamps()
+    end
+
+    create unique_index(:setting, [:id])
+    create unique_index(:setting, [:key])
+    create unique_index(:setting, [:position])
```

Comment:
Inserting settings in existing positions shouldn't really happen. If we ever need to do it though, we could write a function to update all the positions - not that big of a deal since it would only be ran once. The position logic isn't great to be honest, but I think it's good enough for now. I'd like to actually switch to a category system where each category has its set of settings.

I can remove the unique index on the position so we don't have to care much, but it doesn't mean we should start having settings with the same positions - because then, you might show the settings in a different list to the admin panel users every time you reload the page.